app/renderer/components/navigation/navigationBar.js

```diff
@@ -199,80 +283,136 @@ class NavigationBar extends React.Component {
+            commonStyles.navigationButtonContainer,
+            styles.navigationBar__navigationButtonContainer,
+          )}>
+          {this.homeButton}
         </span>
         : null
       }
-      <div className='startButtons'>
-        {
-          !this.props.titleMode
-          ? <span className='bookmarkButtonContainer'>
-            <button data-l10n-id={this.bookmarked ? 'removeBookmarkButton' : 'addBookmarkButton'}
-              className={cx({
-                navigationButton: true,
-                bookmarkButton: true,
-                removeBookmarkButton: this.bookmarked,
-                withHomeButton: getSetting(settings.SHOW_HOME_BUTTON),
-                normalizeButton: true
-              })}
-              onClick={this.onToggleBookmark} />
-          </span>
-          : null
-        }
-      </div>
+      {
+        !this.props.titleMode
+        ? this.bookmarkButtonContainer
+        : null
+      }
       <UrlBar
         titleMode={this.props.titleMode}
         onStop={this.onStop}
         menubarVisible={this.props.menubarVisible}
-        />
+      />
       {
-        isSourceAboutUrl(this.props.location)
-        ? null
-        : <div className='endButtons'>
-          {
-            <PublisherToggle
-              location={this.props.location}
-              locationInfo={this.props.locationInfo}
-              siteSettings={this.props.siteSettings}
-              synopsis={this.props.synopsis}
-            />
-          }
-        </div>
+        !this.props.titleMode && !isSourceAboutUrl(this.props.location)
+        ? this.publisherToggleContainer
+        : null
       }
     </div>
   }
 }
 
+const rightMargin = `calc(${globalStyles.spacing.navbarLeftMarginDarwin} / 2)`
+
 const styles = StyleSheet.create({
-  navigator_wide: {
 
-    // TODO: Refactor navigationBar.js to remove !important
-    maxWidth: '100% !important',
-    marginRight: '0 !important',
-    justifyContent: 'initial !important'
+  navigationBar: {
+    boxSizing: 'border-box',
+    display: 'flex',
+    flexGrow: 1,
+    alignItems: 'center',
+    justifyContent: 'center',
+    fontSize: '20px',
+    minWidth: '0%', // allow the navigationBar to shrink
+    maxWidth: '900px',
+    marginRight: rightMargin,
+    padding: 0,
+    position: 'relative',
+    userSelect: 'none',
+    zIndex: globalStyles.zindex.zindexNavigationBar
+  },
+
+  navigationBar_isDarwin_isFullScreen: {
+    marginRight: 0
+  },
+
+  navigationBar_titleMode: {
+    animation: 'fadeIn 1.2s'
+  },
+
+  navigationBar_wide: {
+    maxWidth: '100%',
+    marginRight: '0',
+    justifyContent: 'initial'
+  },
+
+  // cf: topLevelStartButtonContainer on navigator.js
+  navigationBar__navigationButtonContainer: {
+    // globalStyles.navigationBar.urlbarForm.height + 2px
+    width: '27px'
+  },
+
+  navigationButton: {
+    // cf: https://github.com/brave/browser-laptop/blob/b161b37cf5e9f59be64855ebbc5d04816bfc537b/less/navigationBar.less#L550-L553
```

Comment:
I would leave comments to let possible contributors know what the status of the change is, until everything is converted and refactored to delete not only `navigationBar.less` but also TODO comments finally. The comments help me actually to remember why I made the change there because this PR is getting quite large and larger than as I have expected at first.